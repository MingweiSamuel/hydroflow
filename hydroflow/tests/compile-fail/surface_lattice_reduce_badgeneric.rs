fn main() {
    let mut df = hydroflow::hydroflow_syntax! {
        source_iter([1,2,3,4,5])
            -> lattice_reduce::<'static, usize>()
            -> for_each(|x| println!("Least upper bound: {:?}", x));
    };
    df.run_available();
}
