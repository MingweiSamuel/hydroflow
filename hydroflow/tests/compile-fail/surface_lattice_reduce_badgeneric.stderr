error: `lattice_reduce` should have exactly 0 generic type arguments, actually has 1.
 --> tests/compile-fail/surface_lattice_reduce_badgeneric.rs:4:33
  |
4 |             -> lattice_reduce::<'static, usize>()
  |                                 ^^^^^^^^^^^^^^
